tests/loadtests/vkloadtests/VulkanLoadTestSample.cpp

```diff
@@ -65,11 +65,11 @@ VulkanLoadTestSample::loadMesh(std::string filename,
 vk::PipelineShaderStageCreateInfo
 VulkanLoadTestSample::loadShader(std::string filename,
                                  vk::ShaderStageFlagBits stage,
-                                 std::string modname)
+                                 const char* modname)
```

Comment:
Why are you changing this?

Reply:
Because this function returns `vk::PipelineShaderStageCreateInfo,` which has a pointer to the data in the `std::string`, which is destroyed on return. This was making the load tests crash on my machine. There's in another implementation of `loadShader` in this project that also uses `const char*`, I assume to avoid this problem.

Reply:
This is because the callers are using the default parameter, right? If the caller passed the modname parameter it would still be valid when the function returns and up until the caller exits, by which time the pipeline would be created and it wouldn't matter. Would making the parameter a reference, std::string&, help?

Where is this crashing? Windows? I've run the apps many times on macOs, iOS and Linux and never had a crash. I can't run on Windows because my Windows box doesn't support Vulkan.

Reply:
The pointer would still be invalid if the string is passed by value, like it was in the original version, because the parameter scope is the function IIRC. Passing by reference will extend the lifetime outside this function, but only until the string is destroyed. In this case, if the caller is using the default value, it should become invalid when the caller's scope ends.
Basically the string's lifetime must be as long or longer than that of the `vk::PipelineShaderStageCreateInfo`. By using a pointer to a literal, the string exists in memory for the life of the program, so it's not an issue.
Because this is undefined behaviour, a crash is not guaranteed. This will depend on the compiler optimizations and stdlib implementation of std::string. I had this crash with MSVC in debug, and it was happening later then the driver was trying to use the `vk::PipelineShaderStageCreateInfo` structure. I think MSVC in debug clobbers released memory to help identify these kinds of lifetime bugs, so it crashes were other compilers don't.